cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/preview/LimitingInputFormat.java

```diff
@@ -46,7 +46,10 @@
   public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
     Configuration conf = context.getConfiguration();
     int maxRecords = conf.getInt(MAX_RECORDS, 100);
-    List<InputSplit> splits = createDelegate(conf).getSplits(context);
+    if (delegateFormat == null) {
+      setConf(conf);
```

Comment:
this is pretty messy because it uses the fact that setConf() creates the delegate. 

It's better to change setConf to only set the configuration object, and to change createDelegate() to call the delegate's setConf method if it is Configurable.

The createRecordReader() method should also be changed to call createDelegate(), passing in the Configuration that comes from the context.